cmake/modules/RootBuildOptions.cmake

```diff
@@ -102,37 +102,37 @@ ROOT_BUILD_OPTION(cxx14 OFF "Build using C++14 compatible mode, requires gcc > 4
 #ROOT_BUILD_OPTION(cxx17 OFF "Build using C++17 compatible mode, requires gcc >= 7.1.0 or clang")
 ROOT_BUILD_OPTION(cxxmodules OFF "Compile with C++ modules enabled.")
 ROOT_BUILD_OPTION(davix ON "DavIx library for HTTP/WEBDAV access")
```

Comment:
Can we turn this `OFF` too? It's one of the things that I always have to turn off since I use `-Dfail-on-missing=ON` on my computer.

Reply:
Uhm... I would like to have this since it allows to read files over http with a simple url, which I think it's quite comfortable and widespread.

Reply:
Davix is the 'new'/'recommended' way of accessing http file, why would we disable it?

Reply:
Ok, I wasn't sure what this was, I just have to disable it because I don't have davix on my system and I use fail on missing.

Reply:
> Ok, I wasn't sure what this was, I just have to disable it because I don't have davix on my system and I use fail on missing.

In **this** case, this is the perfect outcome :).  You ought to be warned that something important (but not required) is missing :)

Reply:
I agree that this should stay ON, as it is going to be the recommended way to access files over HTTP.
Guillerme, perhaps you can use -Dbuiltin_davix=ON to overcome your problem?

Reply:
Perhaps, if not there, we should provide a built-in davix in case none is found?

Reply:
@amadio, can you take care once the discussion about this PR is finished and the code is in master? It looks to me that this is not a mere change for davix... Maybe we want to trigger the necessary steps (e.g. inclusion of builtin) for all the packages which are ON by default?  Or even for all packages which are ON? Indeed if I want XYZ which needs TUV, it's understood that I'd like ROOT to take care of TUV if it's not on my system, right?

Reply:
There's no need for change for this option. It is enabled if you don't have it, but I use fail on missing, so I want it to not enable things for me.